hq/app/schedule/IamAudit.scala

```diff
@@ -0,0 +1,64 @@
+package schedule
+
+import com.gu.anghammarad.models.{Notification, AwsAccount => Account}
+import logic.DateUtils
+import model._
+import schedule.IamMessages.createMessage
+import schedule.IamNotifier.createNotification
+import utils.attempt.FailedAttempt
+
+object IamAudit {
+  def makeCredentialsNotification(allCreds: Map[AwsAccount, Either[FailedAttempt, CredentialReportDisplay]]):List[Either[FailedAttempt, Notification]] = {
+    allCreds.toList.map { case (awsAccount, eFCreds) =>
+      eFCreds.map { credsReport =>
+        val outdatedKeys = outdatedKeysInfo(findOldAccessKeys(credsReport))
+        val missingMfa = missingMfaInfo(findMissingMfa(credsReport))
+        val message = createMessage(outdatedKeys, missingMfa)
+        createNotification(Account(awsAccount.id), message)
+      }
+    }
+  }
+
+  def findOldAccessKeys(credsReport: CredentialReportDisplay): CredentialReportDisplay = {
+    val filteredMachines = credsReport.machineUsers.filter(user => hasOutdatedKey(List(user.key1, user.key2)))
+    val filteredHumans = credsReport.humanUsers.filter(user => hasOutdatedKey(List(user.key1, user.key2)))
+    credsReport.copy(machineUsers = filteredMachines, humanUsers = filteredHumans)
+  }
+
+  def hasOutdatedKey(keys: List[AccessKey]): Boolean = keys.exists(key => DateUtils.dayDiff(key.lastRotated).getOrElse(1L) > 90L)
+
+  def findMissingMfa(credsReport: CredentialReportDisplay): CredentialReportDisplay = {
```

Comment:
this function is called `findMissingMfa` but it seems to also do something to the machine users?

Reply:
Yes, it's filtering out all machine users as they cannot have Mfa. I've improved the naming to make that clearer. If there's a better way of doing this, please let me know! Thanks😀